### DISCUSSION TITLE
Integrating local webcams as streaming sources

### DISCUSSION THREAD

**Discussion by maxgrrs** [pedroSG94/RootEncoder#1382] [Q&A] [answered]
Hey Pedro,

for my application it would be nice if I can integrate local rtsp webcams in the stream.
So instead of switching between front and back camera I can choose between several camera sources.

I would like to use only one interface that is responsible for handling the overlay and the streaming.

Is it possible to integrate the stream from the rtsp camera with your library, so that I can apply the same opengl filters and just use the same start/stop stream/preview functions?

Regards, Max

**Comment by pedroSG94** [comment, marked as answer]
Hello,

Currently it is not possible without modify the library but I added a PR 2 days ago that allow this.
https://github.com/pedroSG94/RootEncoder/pull/1381

This PR allow create a class that extend from VideoSource where you can use any source rendering a SurfaceTexture.
I will merge this PR soon. Meanwhile, you can compile the project using that PR and create your own VideoSource class to test.

**Reply by maxgrrs** [reply]
Hey Pedro,
sounds great.
Thanks for fast response, Max
